src/formatDate.js

```diff
@@ -50,7 +50,33 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const arrDate = date.split(fromFormat[3]);
+  const arrNew = [];
+
+  for (const key of toFormat.slice(0, 3)) {
+    let oldIndex = '';
+    let newElement = '';
+
+    if (key === 'YY' && fromFormat.includes('YYYY')) {
+      oldIndex = fromFormat.indexOf('YYYY');
+      newElement = arrDate[oldIndex].slice(2);
+    } else if (key === 'YYYY' && fromFormat.includes('YY')) {
+      oldIndex = fromFormat.indexOf('YY');
+
+      if (arrDate[oldIndex] < 30) {
+        newElement = `20` + `${arrDate[oldIndex]}`;
+      } else {
+        newElement = `19` + `${arrDate[oldIndex]}`;
+      }
+    } else {
+      oldIndex = fromFormat.indexOf(key);
+      newElement = arrDate[oldIndex];
+    }
```

Comment:
avoid such nesting `else if` conditions
This code is hard to read and maintain

Reply:
do your really need all this code inside loop?
you can use different loops for your purposes